### PR TITLE
Switch to Microsoft.Data.SqlClient

### LINKED ISSUE #15636: Move to Microsoft.Data.SqlClient
Now that the [preview of Microsoft.Data.SqlClient is available in NuGet](https://www.nuget.org/packages/Microsoft.Data.SqlClient/), we should be able to use it.

### PR DESCRIPTION
**Pull request by ajcvickers**
Fixes #15636

### CODE REVIEW

File: eng/Versions.props [hunk 1]
```diff
@@ -27,6 +27,7 @@
     <StyleCopAnalyzersPackageVersion>1.1.1-beta.61</StyleCopAnalyzersPackageVersion>
     <SystemInteractiveAsyncPackageVersion>3.2.0</SystemInteractiveAsyncPackageVersion>
     <BenchmarkDotNetPackageVersion>0.11.3</BenchmarkDotNetPackageVersion>
+    <MicrosoftDataSqlClientPackageVersion>1.0.19128.1-Preview</MicrosoftDataSqlClientPackageVersion>
```
Comment (bricelam) [inline eng/Versions.props]: 🙄 SemVer much?

**Comment by natemcmaster** [comment]
Heads up @aspnet/build - this will break aspnet/AspNetCore once merged since our dependency on System.Data.SqlClient is transitive through EF.

File: eng/Versions.props (hunk 1 above)
Comment (divega) [reply eng/Versions.props]: Agreed. Please file an issue on the SqlClient repo.

**Comment by ilmax** [comment]
Just curious about this Microsoft.Data.SqlClient, is it a copy of System.Data.SqlClient at a given point in time or is it a rewrite from the ground up?

**Comment by smitpatel** [comment]
@ilmax - It is rewrite from the ground up.

**Comment by divega** [comment]
@ilmax, @smitpatel It is a copy. Currently a superset as fixes in System.Data.SqlClient done in CoreFx are being ported to the Microsoft.Data.SqlClient, and it contains a few new features. See https://devblogs.microsoft.com/dotnet/introducing-the-new-microsoftdatasqlclient/ for more information.

**Comment by ilmax** [comment]
@divega nice, thanks for the link! One last question. I know @Wraith2 made several perf related changes in System.Data.SqlClient, are we gonna benefit from these changes in M.D.S as well (if applicable)?

File: eng/Versions.props [hunk 2]
```diff
@@ -42,7 +43,6 @@
     <MicrosoftNETCorePlatformsPackageVersion>3.0.0-preview6.19264.9</MicrosoftNETCorePlatformsPackageVersion>
     <SystemCollectionsImmutablePackageVersion>1.6.0-preview6.19264.9</SystemCollectionsImmutablePackageVersion>
     <SystemComponentModelAnnotationsPackageVersion>4.6.0-preview6.19264.9</SystemComponentModelAnnotationsPackageVersion>
-    <SystemDataSqlClientPackageVersion>4.7.0-preview6.19264.9</SystemDataSqlClientPackageVersion>
```
Comment (bricelam) [inline eng/Versions.props]: Remove from Version.Details.xml too

**Comment by divega** [comment]
@ilmax the SqlClient team has ported all of @Wraith2's optimizations to the new codebase.

**Comment by ajcvickers** [comment]
This is currently blocked until I can get https://github.com/aspnet/AspNetCore/pull/10300 merged. So far I have failed to get the ASP.NET build to pass.

**Comment by dougbu** [comment]
This isn't blocked anymore. Clearing that label

**Pull request merged**